editor/inserter/menu.js

```diff
@@ -225,7 +225,20 @@ class InserterMenu extends wp.element.Component {
 	render() {
 		const { position = 'top', instanceId } = this.props;
 		const visibleBlocksByCategory = this.getVisibleBlocksByCategory( wp.blocks.getBlockTypes() );
+		const visualEditorHeight = document.querySelector( '.editor-visual-editor' ).clientHeight;
+		const heightOccupied =
+			document.getElementById( 'wpadminbar' ).clientHeight +
+			document.querySelector( '.editor-header' ).clientHeight +
+			document.querySelector( '.editor-inserter__toggle' ).clientHeight;
```

Comment:
If I click "Insert" in the editor bar, then click the inserter in post content, what node is returned by the query selector here? On the first render of the inserter in post content, it's actually the toggle from the header bar, which isn't correct.

Another one: If we decided to change the selector here to `.editor-inserter__menu`, what is the node on the first render? It wouldn't match anything (`null`) because the element hasn't yet been mounted yet. We're relying on the parent component already having mounted its toggle button, which is not a guarantee and is easy to overlook in refactoring.

I generally try to avoid component DOM querying because of these and other complexities. The `render` shouldn't assume that anything is present in the DOM, nor that it's the only instance of the component. To my previous comment, if DOM access is needed, it should be done via [`ref`](https://facebook.github.io/react/docs/refs-and-the-dom.html).